### DISCUSSION TITLE
Is Pocketbase better suited for a CSR website than a SSR website ? (overall speed)

### DISCUSSION THREAD

**Discussion by madupuis90** [pocketbase/pocketbase#3012] [Q&A] [answered]
Hi, I was looking at using Pocketbase and its authentication features and I found that the JS SDK is very well done and makes authentication super easy to handle on the client site. 

Then I decided to use SSR, but I realized that I need to create a new instance of PocketBase for every request (which is normal), but it made me wonder if the speed gain from SSR is offsetted by this operation. 

Basically I'm wondering if anyone has experience in this domain and provide me with some insight. Will SSR + Pocketbase on every request end up slower than creating an instance in the browser and using CSR ? (I know this is hard to answer because it depends on the site)

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand your concerns.

If you worry about the JS SDK object creation, then you shouldn't really because it won't have any effect on the performance of your application.

What would have effect would be the network requests between the node server and PocketBase.

> Will SSR + Pocketbase on every request end up slower than creating an instance in the browser and using CSR ?

I can't answer this, but my recommendation is to not make your choice between SPA/CSR and SSR solely based on some eventual performance gains.

Use SSR (or any additional server-side handling) to help with the SEO (_if your app is behind an auth this doesn't matter_) OR if you intend to write custom business logic and don't feel comfortable using PocketBase as Go framework (_with the next release there will be option to [bind to the app hooks with JS](https://github.com/pocketbase/pocketbase/discussions/2825#discussioncomment-6376210)_).

Use SPA/CSR if the above is not the case because it will be easier to develop, deploy and less likely to mess up the security. SPA/CSR can be as performant as SSR. Modern bundlers like vite allow automatic creation of chunks based on dynamic `import()` statements so you'll not ship 1 big blob to your clients.

I personally prefer using SPA/CSR because it is easier for me to reason and maintain. As a bonus, when using Go it can be also embedded easily with the builtin Go tools in the final executable.
